src/Compilers/Core/Portable/CommandLine/EditorConfig.cs

```diff
@@ -0,0 +1,206 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Diagnostics;
+using System.IO;
+using System.Threading;
+using System.Text.RegularExpressions;
+
+namespace Microsoft.CodeAnalysis
+{
+    /// <summary>
+    /// Represents a single EditorConfig file, see http://editorconfig.org for details about the format.
+    /// </summary>
+    internal sealed class EditorConfig
+    {
+        // Matches EditorConfig section header such as "[*.{js,py}]", see http://editorconfig.org for details
+        private static readonly Regex s_sectionMatcher = new Regex(@"^\s*\[(([^#;]|\\#|\\;)+)\]\s*([#;].*)?$", RegexOptions.Compiled);
```

Comment:
>s_sectionMatcher [](start = 38, length = 16)

Consider adding some examples and documentation: I see that `#` is the beginning of comments. It would be good to indicate that `;` also mean comments.